lib/buildpack/compile/dotnet_cli.rb

```diff
@@ -0,0 +1,88 @@
+# Encoding: utf-8
+# ASP.NET Core Buildpack
+# Copyright 2014-2016 the original author or authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative '../sdk_info'
+require_relative '../app_dir'
+
+module AspNetCoreBuildpack
+  class DotnetCli
+    include SdkInfo
+
+    def initialize(build_dir, installers)
+      @build_dir = build_dir
+      @installers = installers
+      @app_dir = AppDir.new(@build_dir)
+      @shell = AspNetCoreBuildpack.shell
+    end
+
+    def restore(out)
+      setup_shell_environment
+
+      if msbuild?(@build_dir)
+        @app_dir.project_paths.each do |project|
+          cmd = "bash -c 'cd #{@build_dir}; dotnet restore #{project}'"
+          @shell.exec(cmd, out)
+        end
+      else
+        project_list = @app_dir.project_paths.join(' ')
+        cmd = "bash -c 'cd #{@build_dir}; dotnet restore #{project_list}'"
+        @shell.exec(cmd, out)
+      end
+    end
+
+    def publish(out)
+      setup_shell_environment
+
+      main_project = @app_dir.main_project_path
+      raise 'No project found to build' if main_project.nil?
+
+      publish_dir = File.join(@build_dir, '.cloudfoundry', 'dotnet_publish')
+      FileUtils.mkdir_p(publish_dir)
+
+      cmd = "bash -c 'cd #{@build_dir}; dotnet publish #{main_project} -o #{publish_dir} -c #{publish_config}'"
```

Comment:
This will always fail to compile for self-contained projects with multiple RIDs (most users writing self-contained applications will specify some Windows RID + the Ubuntu RID).  We should specify the RID `ubuntu.14.04-x64` here until cflinuxfs2 stack changes to a different Ubuntu version if the application is self-contained.

Since we can't specify a RID for portable apps, we'll need to look at the csproj file and find the `<PropertyGroup>` element(s) and check for a `<RuntimeIdentifiers>` element inside that.  If it exists, the project is self-contained.